wandb/beta/workflows.py

```diff
@@ -0,0 +1,266 @@
+    elif isinstance(path_or_obj, str):
+        if os.path.isdir(path_or_obj):
+            return artifact.add_dir(path_or_obj)
+        elif os.path.isfile(path_or_obj):
+            return artifact.add_file(path_or_obj)
+        else:
+            with artifact.new_file(name) as f:
+                f.write(json.dumps(path_or_obj, sort_keys=True))
+    else:
+        raise ValueError(
+            f"Expected `path_or_obj` to be instance of `ArtifactEntry`, `WBValue`, or `str, found {type(path_or_obj)}"
+        )
+
+
+def _log_artifact_version(
+    name: str,
+    type: str,
+    entries: Dict[str, Union[str, ArtifactEntry, data_types.WBValue]],
+    aliases: Optional[Union[str, List[str]]] = None,
+    description: Optional[str] = None,
+    metadata: Optional[dict] = None,
+    project: Optional[str] = None,
+    scope_project: Optional[bool] = None,
+    job_type: str = "auto",
+) -> ArtifactInterface:
+    """Creates an artifact, populates it, and logs it with a run.
+    If a run is not present, we create one.
+
+    Args:
+        name: `str` - name of the artifact. If not scoped to a project, name will be suffixed by "-{run_id}".
+        type: `str` - type of the artifact, used in the UI to group artifacts of the same type.
+        entries: `Dict` - dictionary containing the named objects we want added to this artifact.
+        description: `str` - text description of artifact.
+        metadata: `Dict` - users can pass in artifact-specific metadata here, will be visible in the UI.
+        project: `str` - project under which to place this artifact.
+        scope_project: `bool` - if True, we will not suffix `name` with "-{run_id}".
+        job_type: `str` - Only applied if run is not present and we create one.
+            Used to identify runs of a certain job type, i.e "evaluation".
+
+    Returns:
+        ArtifactInterface
+
+    """
+    if wandb.run is None:
+        run = wandb.init(
+            project=project, job_type=job_type, settings=wandb.Settings(silent="true")
+        )
+    else:
+        run = wandb.run
+
+    if not scope_project:
+        name = f"{name}-{run.id}"
+
+    if metadata is None:
+        metadata = {}
+
+    art = wandb.Artifact(name, type, description, metadata, False, None)
+
+    for path in entries:
+        _add_any(art, entries[path], path)
+
+    # "latest" should always be present as an alias
+    aliases = wandb.util._resolve_aliases(aliases)
+    run.log_artifact(art, aliases=aliases)
+
+    return art
+
+
+def log_model(
+    model_obj: Any,
+    name: str = "model",
+    aliases: Optional[Union[str, List[str]]] = None,
+    description: Optional[str] = None,
+    metadata: Optional[dict] = None,
+    project: Optional[str] = None,
+    scope_project: Optional[bool] = None,
+    **kwargs: Dict[str, Any],
+) -> "_SavedModel":
+    """Logs a model object to enable model-centric workflows in the UI.
+    Supported frameworks include PyTorch, Keras, Tensorflow, Scikit-learn, etc.
```

Comment:
in the args description you specify that the model obj can be either `PyTorch, Keras, Tensorflow, Scikit-learn` but here you also have `etc.` probably want to resolve the inconsistency?

Reply:
Those are going to be supported as of now, but we will most likely add more frameworks in the future, so keeping the method description flexible.